### PR TITLE
Move to all millis

### PR DESCRIPTION
**Pull request by gkorland**

### CODE REVIEW

File: docs/commands.md [hunk 1]
```diff
@@ -133,13 +133,13 @@ If this command is used to add data to an existing timeseries, `retentionSecs` a
 Create a compaction rule.
 
 ```sql
-TS.CREATERULE sourceKey destKey AGGREGATION aggType bucketSizeSeconds
+TS.CREATERULE sourceKey destKey AGGREGATION aggType bucketSize
```
Comment (otnielvh) [inline docs/commands.md]: `bucketByTime` might be more representative. Size usually refers to memory

File: src/config.c [hunk 2]
```diff
@@ -39,7 +39,7 @@ int ReadConfig(RedisModuleString **argv, int argc) {
         printf("loaded default retention policy: %lld \n", TSGlobalConfig.retentionPolicy);
         TSGlobalConfig.hasGlobalConfig = TRUE;
     } else {
-        TSGlobalConfig.retentionPolicy = RETENTION_DEFAULT_SECS;
+        TSGlobalConfig.retentionPolicy = RETENTION_DEFAULT;
```
Comment (otnielvh) [inline src/config.c]: how about `DEFAULT_RETENTION_TIME`

Comment (gkorland) [reply src/config.c]: :+1: renamed to RETENTION_TIME_DEFAULT like other defaults

File: docs/commands.md (hunk 1 above)
Comment (gkorland) [reply docs/commands.md]: maybe we should change it to just "interval"?

File: docs/commands.md [hunk 3]
```diff
@@ -340,7 +340,7 @@ TS.INFO key
 TS.INFO temperature:2:32
  1) lastTimestamp
  2) (integer) 1548149279
- 3) retentionSecs
+ 3) retentionTime
```
Comment (K-Jo) [inline docs/commands.md]: @gkorland just checking here if this is not a replace all but effectively in the code

Comment (gkorland) [reply docs/commands.md]: both :) replace all and effectively in the code

File: src/tests/test_module.py [hunk 4]
```diff
@@ -613,9 +613,9 @@ def test_downsampling_rules(self):
     def test_automatic_timestamp(self):
         with self.redis() as r:
             assert r.execute_command('TS.CREATE', 'tester')
-        curr_time = int(time.time())
+        curr_time = int(time.time()*1000)
         r.execute_command('TS.ADD', 'tester', '*', 1)
-        result = r.execute_command('TS.RANGE', 'tester', 0, int(time.time()))
+        result = r.execute_command('TS.RANGE', 'tester', 0, int(time.time() * 1000))
         # test time difference is not more than 5 second
         assert result[0][0] - curr_time <= 5
```
Comment (danni-m) [inline src/tests/test_module.py]: @gkorland 
```suggestion
        assert result[0][0] - curr_time <= 5000
```

File: src/tests.c [hunk 5]
```diff
@@ -11,21 +11,21 @@
 MU_TEST(test_valid_policy) {
     SimpleCompactionRule* parsedRules;
     size_t rulesCount;
-    int result = ParseCompactionPolicy("max:1m:1h;min:10s:5d:10d;avg:2h:10d;avg:3d:100d", &parsedRules, &rulesCount);
+    int result = ParseCompactionPolicy("max:1ms:1h;min:10s:5d:10d;avg:2h:10d;avg:3d:100d", &parsedRules, &rulesCount);
```
Comment (danni-m) [inline src/tests.c]: it doesn't parse `ms`, actually this is an issue after the all-millis fix.
The compacation policy is actually parsing it into seconds normalized value and not millis-normallized.
Its better to decide now so we won't need to break it in the future. Lets make the parsing into millis.

File: docs/commands.md (hunk 1 above)
Comment (danni-m) [reply docs/commands.md]: @gkorland do you think its representative of the meaning? its more a time-bucket ("truncated time").
I think we can go simple and just call this `timeBucket`

File: src/tests/test_module.py (hunk 4 above)
Comment (gkorland) [reply src/tests/test_module.py]: @danni-m I would have leave it at 5ms just change the comment, 5ms should be enough diff

File: docs/commands.md (hunk 1 above)
Comment (gkorland) [reply docs/commands.md]: done

File: src/tests/test_module.py (hunk 4 above)
Comment (gkorland) [reply src/tests/test_module.py]: fixed comment to millisecond

File: src/tests.c (hunk 5 above)
Comment (gkorland) [reply src/tests.c]: fixed it to parse to m to ms and M to minute

**Pull request merged**